plugin_tooling/src/com/github/rustdt/tooling/AbstractRustBuildOutputLineParser.java

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
```

Comment:
RustBuildOutputParser.java contained code for parsing error output from cargo. Since the error output from cargo is left unchanged in rustc 1.12, I wanted to re-use this code. Since the code for parsing cargo output was a special case of regexp-based parsing, I moved all the regexp-based code out of RustBuildOutputParser.java. This file contains the general, abstract code, CargoRustBuildOutputLineParser.java contains the cargo-specific code (that I also wanted to use in the new parser) and ClassicRustcRustBuildOutput**Line**Parser.java contains the regexp-based code that is still used in the "old" parser.